File: app/src/main/java/com/amperas17/mangaedenapp/api/responseprovider/MangaListResponseProvider.java
package com.amperas17.mangaedenapp.api.responseprovider;

import com.amperas17.mangaedenapp.MangaEdenApp;
import com.amperas17.mangaedenapp.api.MangaApiHelper;
import com.amperas17.mangaedenapp.model.manga.MangaListResponse;
import com.amperas17.mangaedenapp.utils.Caller;

import retrofit2.Call;

public class MangaListResponseProvider extends BaseResponseProvider<MangaListResponse, Integer> {

    public MangaListResponseProvider(Caller<MangaListResponse> caller) {
        super(caller);
    }

    @Override
    public Call<MangaListResponse> initCall(Integer... args) {
        return MangaEdenApp.getMangaApi().getMangaList(MangaApiHelper.ENGLISH, 0);
    }
}

File: app/src/main/java/com/amperas17/mangaedenapp/api/responseprovider/MangaListResponseProvider.kt
package com.amperas17.mangaedenapp.api.responseprovider

import com.amperas17.mangaedenapp.api.MangaApiHelper
import com.amperas17.mangaedenapp.model.manga.MangaListResponse
import com.amperas17.mangaedenapp.utils.Caller

import retrofit2.Call


class MangaListResponseProvider(caller: Caller<MangaListResponse>) : BaseResponseProvider<MangaListResponse, String>(caller) {

    override fun initCall(vararg args: String): Call<MangaListResponse> {
        return mangaApi.getMangaList(MangaApiHelper.ENGLISH, 0)
    }
}
